### PR TITLE
3577 smb packer scripts

### PR DESCRIPTION
**Pull request by ilija-lazoroski**
# What does this PR do?

Fixes part of #3577 .

Add any further explanations here.

## PR Checklist
* [x] Have you added an explanation of what your changes do and why you'd like to include them?
* [ ] Is the TravisCI build passing?
* [x] Was the CHANGELOG.md updated to reflect the changes?
* [x] Was the documentation framework updated to reflect the changes?
* [x] Have you checked that you haven't introduced any duplicate code?

## Testing Checklist

* [x] Added relevant unit tests?
* [x] Do all unit tests pass?
* [x] Do all end-to-end tests pass?
* [x] Any other testing performed?
    > Tested by {Running the Monkey locally with relevant config/running Island/...}
* [x] If applicable, add screenshots or log transcripts of the feature working

### CODE REVIEW

**Review by mssalvatore — APPROVED** [review]
Makes sense to me. I'd like to know what @VakarisZ thinks.

File: envs/monkey_zoo/packer/setup_mimikatz_15.yml [hunk 1]
```diff
@@ -0,0 +1,36 @@
+---
+- name: Create a new user
```
Comment (VakarisZ) [inline envs/monkey_zoo/packer/setup_mimikatz_15.yml:2]: These names are not consistent. Either way, this looks like it creates an image in GCP, not user

File: envs/monkey_zoo/packer/smb.pkr.hcl [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+    default = "packer_user"
+}
+variable "packer_user_password" {
+    type = string
+    default = "Passw0rd"
+}
+
+
+
+source "googlecompute" "mimikatz-15" {
```
Comment (VakarisZ) [inline envs/monkey_zoo/packer/smb.pkr.hcl:43]: If packer scripts are **also documentation about the machine**, we should add a description that explains the reason why we need this machine, and what's it used for.

File: envs/monkey_zoo/packer/setup_mimikatz_15.yml [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+          - "Remote Desktop Users"
+
+    - name: Disable SMBv1
+      ansible.windows.win_optional_feature:
+        name: SMB1Protocol
+        state: absent
+
+    - name: Allow port 445 SMB
+      win_command:
+        cmd: netsh advfirewall firewall add rule name="Allow Port 445" dir=in action=allow protocol=TCP localport=445
```
Comment (VakarisZ) [inline envs/monkey_zoo/packer/setup_mimikatz_15.yml:26]: We need to disable powershell remoting ports as well?

File: envs/monkey_zoo/packer/smb.pkr.hcl [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+    disk_size = 50
+    machine_type = "${var.machine_type}"
+    account_file = "${var.account_file}"
+    communicator = "winrm"
+    winrm_username = "${var.packer_username}"
+    winrm_password = "${var.packer_user_password}"
+    winrm_insecure = true
+    winrm_use_ssl = true
+    metadata = {
+      sysprep-specialize-script-cmd = "winrm quickconfig -quiet & net user packer_user Passw0rd /add & net localgroup administrators packer_user /add & winrm set winrm/config/service/auth @{Basic=\"true\"}"
```
Comment (VakarisZ) [inline envs/monkey_zoo/packer/smb.pkr.hcl:57]: Does the packer user get deleted afterwards?

File: envs/monkey_zoo/terraform/firewalls.tf [hunk 5]
```diff
@@ -323,9 +323,9 @@ resource "google_compute_firewall" "deny-rdp64-rdp65-to-others" {
   source_tags = ["rdp-64", "rdp-65"]
 }
 
-// We are disabling PowerShell because we want only RDP to run on these machines
+// We are disabling PowerShell because we want only RDP\SMB to run on these machines
 // and we can't do it via Packer because it uses WinRM to configure the instances
```
Comment (VakarisZ) [inline envs/monkey_zoo/terraform/firewalls.tf:327]: ? We can't run a powershell command using packer? GCP Firewall rules are meant to describe the communication between networks, not disabling services on specific machines.

**Review by VakarisZ — CHANGES_REQUESTED** [review]
Small nit-picks, except firewall rules. We shouldn't create global firewall rules for one instance.

File: envs/monkey_zoo/packer/setup_mimikatz_15.yml (hunk 3 above)
Comment (ilija-lazoroski) [reply envs/monkey_zoo/packer/setup_mimikatz_15.yml:34]: Talked about this via webex

File: envs/monkey_zoo/packer/smb.pkr.hcl (hunk 2 above)
Comment (ilija-lazoroski) [reply envs/monkey_zoo/packer/smb.pkr.hcl:43]: Packer scripts are configuration about machines, not documentation. For documentation we have zoo_network.

File: envs/monkey_zoo/terraform/firewalls.tf (hunk 5 above)
Comment (ilija-lazoroski) [reply envs/monkey_zoo/terraform/firewalls.tf:327]: Maybe we can write a different issue about that and leave this as workaround.

**Comment by VakarisZ** [comment]
Opened a new issue to add cleanup script that disables powershell remoting: https://github.com/guardicore/monkey/issues/3583

**Pull request merged**